lighthouse-core/gather/gatherers/link-elements.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer');
+const getElementsInDocumentString = require('../../lib/page-functions').getElementsInDocumentString;
+
+class LinkElements extends Gatherer {
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   * @return {Promise<LH.Artifacts['LinkElements']>}
+   */
+  async afterPass(passContext) {
+    const driver = passContext.driver;
+
+    // TODO(phulce): merge this with the main resource header values too
```

Comment:
> merge this with the main resource header values too

what does this mean?

Reply:
you can also specify `Link` values in the header of the main document

Reply:
> you can also specify Link values in the header of the main document

ah, good point. `LinkElements` is kind of a confusing name in that case, but maybe that's ok?